drivers/sensor/scd30/scd30.c

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright (c) 2021 Advanced Climate Systems
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#define DT_DRV_COMPAT sensirion_scd30
+
+#include <device.h>
+#include <drivers/i2c.h>
+#include <kernel.h>
+#include <drivers/sensor.h>
+#include <sys/__assert.h>
+#include <sys/byteorder.h>
+#include <sys/crc.h>
+#include <logging/log.h>
+LOG_MODULE_REGISTER(SCD30, CONFIG_SCD30_LOG_LEVEL);
+
+#include "scd30.h"
+
+static const struct scd30_config scd30_0_cfg = {
+	.bus_name = DT_INST_BUS_LABEL(0),
+	.base_address = DT_INST_REG_ADDR(0),
+};
+
+struct scd30_data scd30_0_data;
+
+static inline const struct device *scd30_i2c_device(const struct device *dev)
+{
+	const struct scd30_data *ddp = dev->data;
+
+	return ddp->bus;
+}
+
+static inline uint8_t scd30_i2c_address(const struct device *dev)
+{
+	const struct scd30_config *dcp = dev->config;
+
+	return dcp->base_address;
+}
+
+/**
+ * @brief Take the received  word (16 bit), if the data is valid, add it to dst
+ *
+ * @param data data received from sensor
+ * @param dst buffer to store data
+ * @return int
+ */
+static int scd30_fill_data_buf(uint8_t *data, uint8_t *dst)
+{
+	if (scd30_check_crc(data, 2, data[2])) {
+		return -EIO;
+	}
+
+	dst[0] = data[0];
+	dst[1] = data[1];
+
+	return 0;
+}
+
+uint16_t scd30_bytes_to_uint16_t(const uint8_t *bytes)
+{
+	return (uint16_t)bytes[0] << 8 | (uint16_t)bytes[1];
+}
+
+uint32_t scd30_bytes_to_uint32_t(const uint8_t *bytes)
+{
+	return (uint32_t)bytes[0] << 24 | (uint32_t)bytes[1] << 16 | (uint32_t)bytes[2] << 8 |
+	       (uint32_t)bytes[3];
+}
+
+float scd30_bytes_to_float(const uint8_t *bytes)
+{
+	union {
+		uint32_t u32_value;
+		float float32;
+	} tmp;
+
+	tmp.u32_value = scd30_bytes_to_uint32_t(bytes);
+	return tmp.float32;
+}
+
+static int scd30_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	uint8_t rx_buf[6 * (SCD30_WORD_SIZE + SCD30_CRC8_LEN)] = {};
+	uint8_t data_buf[6 * SCD30_WORD_SIZE] = {};
+	struct scd30_data *data = dev->data;
+	int rc = 0;
+
+	rc = scd30_write_command(dev, SCD30_CMD_GET_DATA_READY);
+	if (rc != 0) {
+		return -EIO;
+	}
+
+	rc = i2c_read(scd30_i2c_device(dev), rx_buf, SCD30_WORD_SIZE + SCD30_CRC8_LEN,
```

Comment:
Please use the `_dt` helpers to avoid boilerplate, e.g. use `i2c_read_dt` here instead.